[18.0][MIG] product_variant_sale_price: Migration to 18.0

NICO-SOLUTIONS opened the pull request:
standard migration

sergio-teruel committed:
[ADD][8.0] product_variant_sale_price: New module

oca-transbot committed:
OCA Transbot updated translations from Transifex

tafaRU committed:
[product_variant_sale_price] Also set fix_price to variant when template is created

[product_variant_sale_price] Set minimun fix_price among variants to ensure consistency with the price shown in the shop

oca-transbot committed:
OCA Transbot updated translations from Transifex

tafaRU committed:
[FIX] product_variant_sale_price: avoid to update list price of all variants

Steps to reproduce:
- create more than one variant
- change fix_price of one variant

Current behavior:
- lst_price of all variants are changed

Expected behavior:
- lst_price of only selected variant is changed

SimoRubi committed:
[MIG] product_variant_sale_price: Migration to 11.0

oca-travis committed:
[UPD] Update product_variant_sale_price.pot

OCA-git-bot committed:
[UPD] README.rst

Commit:
Translated using Weblate (Dutch)

Currently translated at 100,0% (4 of 4 strings)

Translation: product-variant-11.0/product-variant-11.0-product_variant_sale_price
Translate-URL: https://translation.odoo-community.org/projects/product-variant-11-0/product-variant-11-0-product_variant_sale_price/nl/

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

OCA-git-bot committed:
[UPD] README.rst

cubells committed:
[MIG] product_variant_sale_price: Migration to 12.0

oca-travis committed:
[UPD] Update product_variant_sale_price.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-variant-12.0/product-variant-12.0-product_variant_sale_price
Translate-URL: https://translation.odoo-community.org/projects/product-variant-12-0/product-variant-12-0-product_variant_sale_price/

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (6 of 6 strings)

Translation: product-variant-12.0/product-variant-12.0-product_variant_sale_price
Translate-URL: https://translation.odoo-community.org/projects/product-variant-12-0/product-variant-12-0-product_variant_sale_price/pt/

Rad0van committed:
[IMP] product_variant_sale_price: black, isort, prettier

Rad0van committed:
[MIG] product_variant_sale_price: Migration to 13.0

Commit:
[MIG] product_variant_sale_price: Migration to 14.0

Commit:
[IMP] product_variant_sale_price: black, isort, prettier

Commit:
!fixup product variant easy view

oca-travis committed:
[UPD] Update product_variant_sale_price.pot

OCA-git-bot committed:
[UPD] README.rst

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (6 of 6 strings)

Translation: product-variant-14.0/product-variant-14.0-product_variant_sale_price
Translate-URL: https://translation.odoo-community.org/projects/product-variant-14-0/product-variant-14-0-product_variant_sale_price/pt/

yankinmax committed:
[IMP] product_variant_sale_price: pylint

yankinmax committed:
[IMP] product_variant_sale_price: remove redundant super className in Python3

yankinmax committed:
[MIG] product_variant_sale_price: Migration to 15.0

Commit:
[UPD] Update product_variant_sale_price.pot

ajaniszewska-dev committed:
Translated using Weblate (French)

Currently translated at 83.3% (5 of 6 strings)

Translation: product-variant-15.0/product-variant-15.0-product_variant_sale_price
Translate-URL: https://translation.odoo-community.org/projects/product-variant-15-0/product-variant-15-0-product_variant_sale_price/fr/

Noel000 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (6 of 6 strings)

Translation: product-variant-15.0/product-variant-15.0-product_variant_sale_price
Translate-URL: https://translation.odoo-community.org/projects/product-variant-15-0/product-variant-15-0-product_variant_sale_price/ca/

ecino committed:
[IMP] product_variant_sale_price: pre-commit stuff

ecino committed:
[MIG] product_variant_sale_price: Migration to 16.0

ecino committed:
[IMP] product_variant_sale_price: Set price_extra to 0

Apply suggested changes from FrankC013

ecino committed:
FIX tests for working alongside product_variant_name module

Commit:
[UPD] Update product_variant_sale_price.pot

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: product-variant-16.0/product-variant-16.0-product_variant_sale_price
Translate-URL: https://translation.odoo-community.org/projects/product-variant-16-0/product-variant-16-0-product_variant_sale_price/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: product-variant-16.0/product-variant-16.0-product_variant_sale_price
Translate-URL: https://translation.odoo-community.org/projects/product-variant-16-0/product-variant-16-0-product_variant_sale_price/it/

pedroguirao committed:
[MIG][17.0] product_variant_sale_price: Migration to 17.0

FIX

[UPD] Update product_variant_sale_price.pot

[BOT] post-merge updates

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-variant-17.0/product-variant-17.0-product_variant_sale_price
Translate-URL: https://translation.odoo-community.org/projects/product-variant-17-0/product-variant-17-0-product_variant_sale_price/

Translated using Weblate (Italian)

Currently translated at 100.0% (7 of 7 strings)

Translation: product-variant-17.0/product-variant-17.0-product_variant_sale_price
Translate-URL: https://translation.odoo-community.org/projects/product-variant-17-0/product-variant-17-0-product_variant_sale_price/it/

Translated using Weblate (Spanish)

Currently translated at 100.0% (7 of 7 strings)

Translation: product-variant-17.0/product-variant-17.0-product_variant_sale_price
Translate-URL: https://translation.odoo-community.org/projects/product-variant-17-0/product-variant-17-0-product_variant_sale_price/es/

gabe-grinspan committed:
[FIX] product_variant_sale_price: edit Sales Price

[BOT] post-merge updates

NICO-SOLUTIONS committed:
[MIG] product_variant_sale_price: Migration to 18.0

ph-obs commented:
Hi @NICO-SOLUTIONS. May I ask if you are still working on this, or if ther are any updates?

NICO-SOLUTIONS commented:
> Hi @NICO-SOLUTIONS. May I ask if you are still working on this, or if ther are any updates?

Feel free to review...

rousseldenis commented:
/ocabot migration product_variant_sale_price

rousseldenis commented on product_variant_sale_price/__manifest__.py:
```diff
@@ -5,7 +5,7 @@
     "name": "Product Variant Sale Price",
     "summary": "Allows to write fixed prices in product variants",
-    "version": "17.0.1.1.0",
+    "version": "18.0.1.1.0",
```
Could you reset to `18.0.1.0.0`? Thanks

rousseldenis commented on product_variant_sale_price/models/product_product.py:
```diff
@@ -0,0 +1,108 @@
+        """the sale.order.line module calculates the price_unit by adding
+        the value of price_extra and this can generate inconsistencies
+        if the field has old data stored."""
+        for product in self:
```
```suggestion
        self.price_extra = 0.0
```

rousseldenis commented on product_variant_sale_price/models/product_product.py:
```diff
@@ -0,0 +1,108 @@
+        records = super().create(vals)
+        for i, product_tmpl in enumerate(records):
+            single_vals = vals[i] if isinstance(vals, list) else vals
+            product_tmpl._update_fix_price(single_vals)
```
IMHO, this should be improved.

Instead, I would have looked at `_create_variant_ids()` and `_prepare_variant_values()` in order to avoid too much writes as using variant `create()` instead.

rousseldenis replied:
@tafaRU @ecino @carlosdauden

NICO-SOLUTIONS replied:
@rousseldenis please let me know, if the IMP-commit contains what you were thinking of

NICO-SOLUTIONS committed:
[IMP] product_variant_sale_price: set fixed prices at creation of variants & optimize synchronisation

- _create_variant_ids(): Variants receive fix_price directly when created
- write(): Efficiently updates fix_price of variants when `list_price` is changed
- Removes unnecessary _update_fix_price() calls for better performance

matteotognini reviewed (approved):
functional ok

ecino reviewed (approved):
LGTM :+1:

AlienAtSystem commented on product_variant_sale_price/models/product_product.py:
```diff
@@ -0,0 +1,108 @@
+            combination,
+            product_id,
+            add_qty,
+            pricelist,
```
pricelist was removed as a kwarg from _get_combination_info in Odoo18. It needs to be removed or this call will crash

NICO-SOLUTIONS committed:
[FIX] product_variant_sale_price: Remove deprecated `pricelist` kwarg from _get_combination_info

In Odoo 18, the keyword argument pricelist was removed from the _get_combination_info method. This PR removes the parameter from the calls to avoid a TypeError

NICO-SOLUTIONS committed:
[FIX] product_variant_sale_price: Avoid singleton error in _create_variant_ids

- The singleton error in _create_variant_ids will be avoided by iterating over self

jappi00 commented:
LGTM

FrancoMaxime commented on product_variant_sale_price/__manifest__.py:
```diff
@@ -0,0 +1,17 @@
+    "version": "18.0.1.0.0",
+    "category": "Product Management",
+    "website": "https://github.com/OCA/product-variant",
+    "author": "Tecnativa, " "Odoo Community Association (OCA)",
```
```suggestion
    "author": "Tecnativa, Odoo Community Association (OCA)",
```

FrancoMaxime commented on product_variant_sale_price/models/product_product.py:
```diff
@@ -0,0 +1,107 @@
+            for variant in tmpl.product_variant_ids:
+                variant.with_context(skip_update_fix_price=True).write(
+                    {"fix_price": tmpl.list_price}
+                )
```
```suggestion
         tmpl.product_variant_ids.with_context(skip_update_fix_price=True).write(
                    {"fix_price": tmpl.list_price}
                )
```

FrancoMaxime replied:
I think as its the same value for each product_variant_ids, its more optimized to make only one write for all of them. No?

FrancoMaxime commented on product_variant_sale_price/models/product_product.py:
```diff
@@ -0,0 +1,107 @@
+        return res
+
+
+class ProductProduct(models.Model):
```
@rousseldenis Hi Denis, i didn't found an answer in the [contribution guide](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#modules). But should the two classes be split into two files?

NICO-SOLUTIONS replied:
just let me know if you want to have this separated

FrancoMaxime replied:
As you wish, IMO I prefer split into 2 files. But there is not standard rule for that

NICO-SOLUTIONS replied:
@FrancoMaxime 
methods are splitted now in 2 files.

FrancoMaxime reviewed (commented):
Hi @NICO-SOLUTIONS, thanks for all your work. great Job

NICO-SOLUTIONS committed:
[IMP] product_variant_sale_price: optimize fix_price initialization for product variants

ecino commented:
@OCA/product-maintainers Can we merge this one?

NICO-SOLUTIONS committed:
[REF] product_variant_sale_price: split product models and rename test file

- Moved product.template methods to dedicated product_template.py
- Renamed test_product_product.py -> test_product.py since both models are involved in the tests

NICO-SOLUTIONS commented:
@OCA/product-maintainers 
can we merge this one please?

AEstLo commented:
@simahawk do you mind to merge this one?

PaulGoubert reviewed (approved):
Funtionnal review OK. 
thank you Nico-Solutions

simahawk commented:
@NICO-SOLUTIONS could you please rewrite the last commits to include the module name as per our guidelines? 
Thanks!

<img width="738" height="560" alt="image" src="https://github.com/user-attachments/assets/074855c3-063e-4762-a36b-ca45a2d1e2dc" />

NICO-SOLUTIONS commented:
@simahawk 
yeah sure.... rewriting of the mentioned commits done!

AEstLo commented:
so can this be merged? @simahawk 
Thanks in advance!

simahawk commented:
/ocabot merge nobump
